### PR TITLE
COOK-62 Support SSL everywhere

### PR DESCRIPTION
**Pull request by wolf31o2**
- Create SSL keystore for Router
- Use correct keypassword for Auth Server
- Only create keystores if type == JKS
- Move SSL certificate creation to main recipes
- Move tests to correct specs

### CODE REVIEW

File: recipes/gateway.rb [hunk 1]
```diff
@@ -24,6 +24,37 @@
...
 
+# Create a new keystore if SSL is enabled (and we don't already have one in place)
+execute 'create-router-ssl-keystore' do
+  ssl_enabled =
+    if node['cdap']['version'].to_f < 2.5 && node['cdap'].key?('cdap_site') &&
+       node['cdap']['cdap_site'].key?('security.server.ssl.enabled')
+      node['cdap']['cdap_site']['security.server.ssl.enabled']
+    elsif node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('ssl.enabled')
+      node['cdap']['cdap_site']['ssl.enabled']
+    # This one is here for compatibility, but ssl.enabled takes precidence, if set
```
Comment (dereklwood) [inline recipes/gateway.rb]: precedence

File: recipes/security.rb [hunk 2]
```diff
@@ -34,6 +34,63 @@
...
 
+# Create a new keystore if SSL is enabled
+execute 'create-security-server-ssl-keystore' do
+  ssl_enabled =
+    if node['cdap']['version'].to_f < 2.5 && node['cdap'].key?('cdap_site') &&
+       node['cdap']['cdap_site'].key?('security.server.ssl.enabled')
+      node['cdap']['cdap_site']['security.server.ssl.enabled']
+    elsif node['cdap'].key?('cdap_site') && node['cdap']['cdap_site'].key?('ssl.enabled')
+      node['cdap']['cdap_site']['ssl.enabled']
+    # This one is here for compatibility, but ssl.enabled takes precidence, if set
```
Comment (dereklwood) [inline recipes/security.rb]: precedence

**Comment by dereklwood** [comment]
it seems a little wierd to me to use ``type`` as the keystore/cert generation guard.  It seems possible to 1) forget about it, or 2) when using your own cert you are then dependent on the ``not_if File.exist?`` guards to prevent the cookbook from generating a keystore overtop your existing one (unless you also don't set the ``type`` attr since its the default, but that's wierd too).

i'm thinking an explicit ``use_self_signed`` attr or similar would give a more straightforward control over this behavior.  open to discussion.

**Comment by dereklwood** [comment]
suggestion/question for discussion, and couple minor comments.  otherwise :+1:

**Comment by wolf31o2** [comment]
I added the `type` guard only after realizing that CDAP supported no-JKS keystores and the cookbook was written to always create a JKS keystore. Truthfully, we should be setting the correct command based on type. The cookbook's always generated the certificate if one isn't present. I intended this to be a non-breaking change, which is the reason for not introducing a new attribute to control this behavior.

**Pull request merged**